app/services/invoices/preview_service.rb

```diff
@@ -13,13 +15,14 @@ def initialize(customer:, subscription:, applied_coupons: [])
     def call
       return result.forbidden_failure! unless License.premium?
       return result.not_found_failure!(resource: 'customer') unless customer
-      return result.not_found_failure!(resource: 'subscription') unless subscription
+      return result.not_found_failure!(resource: 'subscription') if subscriptions.empty?
+      return result unless currencies_aligned?
 
       @invoice = Invoice.new(
         organization: customer.organization,
         customer:,
         invoice_type: :subscription,
-        currency: subscription.plan&.amount_currency,
```

Comment:
I'm always scared that this `&` was there for a good reason I don't know 🙈 maybe it wasn't tho!

Reply:
We have validation for it in a plan so it should be always there :)